packages/react/src/components/Storage/FileUploader/UploadButton/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { translate } from '@aws-amplify/ui';
+import { UploadButtonProps } from '../types';
+import { Button, VisuallyHidden } from '../../../../primitives';
+
+export function UploadButton({
+  multiple,
+  acceptedFileTypes,
+  onFileChange,
+  className,
+  isLoading,
+  hiddenInput,
+  onClick,
+}: UploadButtonProps): JSX.Element {
+  return (
+    <>
+      <Button
+        disabled={isLoading}
+        className={className}
+        size="small"
+        onClick={onClick}
+        variation="primary"
+      >
+        {translate('Browse files')}
+      </Button>
+      <VisuallyHidden>
+        <input
+          type="file"
+          tabIndex={-1}
+          ref={hiddenInput}
+          onChange={onFileChange}
+          multiple={multiple}
+          accept={acceptedFileTypes?.join()}
+        />
+      </VisuallyHidden>
```

Comment:
Let's move this above the `UploadButton` it shouldn't be a heavy lift and it will allow us to no longer need to pass `hiddenInput`, `onFileChange`, `multiple`, `acceptedFileTypes` in to `UploadButton` which will decrease complexity overall. Should just be as simple as:
```
// in FileUploader.tsx
const uploadButton = (
    <>
      <UploadButton {...CommonProps} />
      <VisuallyHidden>
        <input
          type="file"
          tabIndex={-1}
          ref={hiddenInput}
          onChange={onFileChange}
          multiple={multiple}
          accept={acceptedFileTypes?.join()}
        />
      </VisuallyHidden>
    </>
  )

  if (variation === 'button') {
    return uploadButton;
  } else {
    return (
      <UploadDropZone
        inDropZone={inDropZone}
        onDragEnter={onDragEnter}
        onDragLeave={onDragLeave}
        onDragOver={onDragOver}
        onDragStart={onDragStart}
        onDrop={onDrop}
      >
        <Text className={ComponentClassNames.FileUploaderDropZoneText}>
          {translate('Drop files here or')}
        </Text>
        {uploadButton}
      </UploadDropZone>
    );
  }
```

Reply:
I'm not sure if this will work out. We have to re-use the UploadButton in the `Previewer` component as well. So we are back to passing in refs, or duplicating the `onClick` code in both the `FileUploader and the `Previewer` code. Not sure if there is an elegant way to handle this.